platform/core/src/services/HangingProtocolService/HangingProtocolService.js

```diff
@@ -339,9 +351,19 @@ class HangingProtocolService {
     // Note: this is happening before displaySets are created. Here, displaySet
     // only contains the information of the id of the displaySet to be matched
     // based on some rules
+    const matchedUids = {};
+    this.customAttributeRetrievalCallbacks.alreadyMatched = (metadata) => {
+      return !!matchedUids[metadata.uid];
+    }
+
     stageModel.displaySets.forEach(displaySet => {
-      const { bestMatch } = this._matchImages(displaySet);
+      const { bestMatch, matchingScores } = this._matchImages(displaySet);
```

Comment:
The matchingScores list is already returned, and gets used to allow a hanging protocol to specify a best match other than the 0 match, or to specify the zero match a second time, with alternate parameters.